flash/vision/detection/model.py

```diff
@@ -57,25 +62,44 @@ class ObjectDetector(Task):
     def __init__(
         self,
         num_classes: int,
-        model: Union[str, nn.Module] = "fasterrcnn_resnet50_fpn",
+        backbone: Optional[str] = None,
+        fpn: bool = True,
+        pretrained: bool = True,
+        pretrained_backbone: bool = True,
+        trainable_backbone_layers: int = 3,
         loss=None,
         metrics: Union[Callable, nn.Module, Mapping, Sequence, None] = None,
         optimizer: Type[Optimizer] = torch.optim.Adam,
-        pretrained: bool = True,
-        learning_rate=1e-3,
-        **kwargs,
+        learning_rate: float = 1e-3,
+        **kwargs: Any,
     ):
 
         self.save_hyperparameters()
 
-        if model in _models:
-            model = _models[model](pretrained=pretrained)
-            if isinstance(model, torchvision.models.detection.FasterRCNN):
-                in_features = model.roi_heads.box_predictor.cls_score.in_features
-                head = torchvision.models.detection.faster_rcnn.FastRCNNPredictor(in_features, num_classes)
-                model.roi_heads.box_predictor = head
+        if backbone is None:
+            model = fasterrcnn_resnet50_fpn(
+                pretrained=pretrained,
+                pretrained_backbone=pretrained_backbone,
+                trainable_backbone_layers=trainable_backbone_layers,
+            )
+            in_features = model.roi_heads.box_predictor.cls_score.in_features
+            head = FastRCNNPredictor(in_features, num_classes)
+            model.roi_heads.box_predictor = head
         else:
-            ValueError(f"{model} is not supported yet.")
+            backbone_model, num_features = fetch_fasterrcnn_backbone_and_num_features(
+                backbone,
+                fpn,
+                pretrained_backbone,
+                trainable_backbone_layers,
+                **kwargs,
+            )
+            backbone_model.out_channels = num_features
+            anchor_generator = AnchorGenerator(sizes=((32, 64, 128, 256, 512), ),
```

Comment:
is this fixed or can we allow passing custom anchors?

Reply:
I think let's have it fixed for the first iteration. It aligns with `Flash's` idea of fast prototyping and baselining & User base. We could make it more flexible in the next iteration with support for more params.

Reply:
yes, I agree with goo defaults, my point is to add an extra argument `anchor_generator` which would be None by default which lead to use this one you just wrote, but if a user wants to use other he doe not need to overwrite all

Reply:
Updated!